File: src/main/java/kingdoms/server/handlers/EntitiesRegister.java
package kingdoms.server.handlers;

import cpw.mods.fml.client.registry.RenderingRegistry;
import cpw.mods.fml.common.registry.EntityRegistry;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import kingdoms.client.render.RenderBipedToK;
import kingdoms.server.TaleOfKingdoms;
import kingdoms.server.entities.*;
import net.minecraft.client.model.ModelBiped;
import net.minecraft.entity.Entity;

import java.util.Calendar;
import java.util.Date;

public final class EntitiesRegister
{
    private byte id = 0;
    private int month, day;

    public static final EntitiesRegister INSTANCE = new EntitiesRegister();

    private EntitiesRegister()
    {
        Calendar calendar = Calendar.getInstance();
        calendar.setTime(new Date());
        month = calendar.get(Calendar.MONTH) + 1;
        day = calendar.get(Calendar.DATE);
    }

    public void register()
    {
        setRegister(EntityFisher.class, "Fisher");
        setRegister(EntityLumber.class, "Lumber");
        setRegister(EntityWorkerMember.class, "WorkerMember");
        setRegister(EntityDefendMarker.class, "DefendMark");
        setRegister(EntityMageKeeper.class, "MageKeeper");
        setRegister(EntityGuildMaster.class, "GuildMaster");
        setRegister(EntityInnKeeper.class, "InnKeeper");
        setRegister(EntityHunterKeeper.class, "GuildKeeper");
        setRegister(EntityBuilderKeeper.class, "BuilderKeeper");
        setRegister(EntityBankerKeeper.class, "Banker");
        setRegister(EntityBarracksKeeper.class, "BarracksKeeper");
        setRegister(EntityDefendArcher.class, "DefendArcher");
        setRegister(EntityDefendBandit.class, "DefendBandit");
        setRegister(EntityDefendKnight.class, "DefendKnight");
        setRegister(EntityDefendMage.class, "DefendMage");
        setRegister(EntityDefendPaladin.class, "DefendPaladin");
        setRegister(EntityDefendPriest.class, "DefendPriest");
        setRegister(EntityDefendWarrior.class, "DefendWarrior");
        setRegister(EntityFarmerKeeper.class, "Farmer");
        setRegister(EntityFoodKeeper.class, "FoodKeeper");
        setRegister(EntityQuarry.class, "Quarry");
        setRegister(EntityWeaponKeeper.class, "WeaponKeeper");
        setRegister(EntityGuildMember.class, "GuildMember");
        setRegister(EntityHired.class, "Hired");
        setRegister(EntityForgeKeeper.class, "ForgeKeeper");
        setRegister(EntityHeadCommander.class, "HeadCommander");
        setRegister(EntityKingdomWorker.class, "KingdomWorker");
        setRegister(EntityLibraryKeeper.class, "LibraryKeeper");
        setRegister(EntityLoneTraveller.class, "LoneTraveller");
        setRegister(EntityLostVillager.class, "LostVillager");
        setRegister(EntityMarkerKeeper.class, "Marker");
        setRegister(EntityMarker2Keeper.class, "Marker2");
        setRegister(EntityPriestKeeper.class, "PriestKeeper");
        setRegister(EntityReficulGuardian.class, "ReficulGuardian");
        setRegister(EntityReficulMage.class, "ReficulMage");
        setRegister(EntityReficulSoldier.class, "ReficulSoldier");
        setRegister(EntityShopKeeper.class, "ShopKeeper");
        setRegister(EntityStockKeeper.class, "StockKeeper");
        setRegister(EntityTavernKeeper.class, "TavernKeeper");
        setRegister(EntityVillageMember.class, "VillageMember");

        if ((month == 12 || month == 1) && (day >= 1 && day <= 31))
            setRegister(EntitySantaClaus.class, "SantaClaus");
    }

    public void render()
    {
        setRender(EntityGuildMaster.class, "head");
        setRender(EntityInnKeeper.class, "inn");
        setRender(EntityBuilderKeeper.class, "builder");
        setRender(EntityHunterKeeper.class, "head");
        setRender(EntityBankerKeeper.class, "banker");
        setRender(EntityBarracksKeeper.class, "guardelite");
        setRender(EntityDefendArcher.class, "hunter");
        setRender(EntityDefendBandit.class, "bandit");
        setRender(EntityDefendKnight.class, "knight");
        setRender(EntityDefendMage.class, "wizard");
        setRender(EntityDefendMarker.class, "");
        setRender(EntityDefendPaladin.class, "paladin");
        setRender(EntityDefendPriest.class, "priest");
        setRender(EntityDefendWarrior.class, "warrior");
        setRender(EntityFarmerKeeper.class, "inn");
        setRender(EntityFoodKeeper.class, "food");
        setRender(EntityLumber.class, "foremanlumber");
        setRender(EntityWeaponKeeper.class, "smith");
        setRender(EntityGuildMember.class, "member");
        setRender(EntityHired.class, "guild");
        setRender(EntityForgeKeeper.class, "forge");
        setRender(EntityHeadCommander.class, "headcommander");
        setRender(EntityKingdomWorker.class, "worker");
        setRender(EntityLibraryKeeper.class, "librarian");
        setRender(EntityLoneTraveller.class, "lone");
        setRender(EntityLostVillager.class, "ac");
        setRender(EntityMarkerKeeper.class, "");
        setRender(EntityMarker2Keeper.class, "");
        setRender(EntityPriestKeeper.class, "headpriest");
        setRender(EntityReficulGuardian.class, "reficulGuardian");
        setRender(EntityReficulMage.class, "reficulMage");
        setRender(EntityReficulSoldier.class, "reficulSoldier");
        setRender(EntityShopKeeper.class, "forge");
        setRender(EntityStockKeeper.class, "stock");
        setRender(EntityTavernKeeper.class, "tavern");
        setRender(EntityVillageMember.class, "man1");
        setRender(EntityWorkerMember.class, "worker");
        setRender(EntityMageKeeper.class, "headmage");
        setRender(EntityQuarry.class, "foremanquarry");
        setRender(EntityFisher.class, "fisher");

        if ((month == 12 || month == 1) && (day >= 1 && day <= 31))
            setRender(EntitySantaClaus.class, "santaClaus");
    }

    private void setRegister(Class<? extends Entity> entity, String name)
    {
        EntityRegistry.registerModEntity(entity, name, id++, TaleOfKingdoms.instance, 64, 1, false);
    }

    @SideOnly(Side.CLIENT)
    private void setRender(Class<? extends Entity> entity, String location)
    {
        RenderingRegistry.registerEntityRenderingHandler(entity, new RenderBipedToK(new ModelBiped(), 0.4F, location));
    }
}

File: src/main/java/kingdoms/server/handlers/EntitiesRegister.kt
package kingdoms.server.handlers

import cpw.mods.fml.client.registry.RenderingRegistry
import cpw.mods.fml.common.registry.EntityRegistry
import cpw.mods.fml.relauncher.Side
import cpw.mods.fml.relauncher.SideOnly
import kingdoms.client.render.RenderBipedToK
import kingdoms.server.TaleOfKingdoms
import kingdoms.server.entities.*
import net.minecraft.client.model.ModelBiped
import net.minecraft.entity.Entity
import java.util.*

object EntitiesRegister
{
    private var id: Int = 0
    private val isNewYear: Boolean
    
    init
    {
        val calendar = Calendar.getInstance()
        calendar.time = Date()
        val month = calendar.get(Calendar.MONTH) + 1
        val day = calendar.get(Calendar.DATE)
        isNewYear = (month == 12 || month == 1) && (day in 1..31)
    }

    fun register()
    {
        setRegister(EntityFisher::class.java, "Fisher")
        setRegister(EntityLumber::class.java, "Lumber")
        setRegister(EntityWorkerMember::class.java, "WorkerMember")
        setRegister(EntityDefendMarker::class.java, "DefendMark")
        setRegister(EntityMageKeeper::class.java, "MageKeeper")
        setRegister(EntityGuildMaster::class.java, "GuildMaster")
        setRegister(EntityInnKeeper::class.java, "InnKeeper")
        setRegister(EntityHunterKeeper::class.java, "GuildKeeper")
        setRegister(EntityBuilderKeeper::class.java, "BuilderKeeper")
        setRegister(EntityBankerKeeper::class.java, "Banker")
        setRegister(EntityBarracksKeeper::class.java, "BarracksKeeper")
        setRegister(EntityDefendArcher::class.java, "DefendArcher")
        setRegister(EntityDefendBandit::class.java, "DefendBandit")
        setRegister(EntityDefendKnight::class.java, "DefendKnight")
        setRegister(EntityDefendMage::class.java, "DefendMage")
        setRegister(EntityDefendPaladin::class.java, "DefendPaladin")
        setRegister(EntityDefendPriest::class.java, "DefendPriest")
        setRegister(EntityDefendWarrior::class.java, "DefendWarrior")
        setRegister(EntityFarmerKeeper::class.java, "Farmer")
        setRegister(EntityFoodKeeper::class.java, "FoodKeeper")
        setRegister(EntityQuarry::class.java, "Quarry")
        setRegister(EntityWeaponKeeper::class.java, "WeaponKeeper")
        setRegister(EntityGuildMember::class.java, "GuildMember")
        setRegister(EntityHired::class.java, "Hired")
        setRegister(EntityForgeKeeper::class.java, "ForgeKeeper")
        setRegister(EntityHeadCommander::class.java, "HeadCommander")
        setRegister(EntityKingdomWorker::class.java, "KingdomWorker")
        setRegister(EntityLibraryKeeper::class.java, "LibraryKeeper")
        setRegister(EntityLoneTraveller::class.java, "LoneTraveller")
        setRegister(EntityLostVillager::class.java, "LostVillager")
        setRegister(EntityMarkerKeeper::class.java, "Marker")
        setRegister(EntityMarker2Keeper::class.java, "Marker2")
        setRegister(EntityPriestKeeper::class.java, "PriestKeeper")
        setRegister(EntityReficulGuardian::class.java, "ReficulGuardian")
        setRegister(EntityReficulMage::class.java, "ReficulMage")
        setRegister(EntityReficulSoldier::class.java, "ReficulSoldier")
        setRegister(EntityShopKeeper::class.java, "ShopKeeper")
        setRegister(EntityStockKeeper::class.java, "StockKeeper")
        setRegister(EntityTavernKeeper::class.java, "TavernKeeper")
        setRegister(EntityVillageMember::class.java, "VillageMember")

        if (isNewYear)
            setRegister(EntitySantaClaus::class.java, "SantaClaus")
    }

    fun render()
    {
        setRender(EntityGuildMaster::class.java, "head")
        setRender(EntityInnKeeper::class.java, "inn")
        setRender(EntityBuilderKeeper::class.java, "builder")
        setRender(EntityHunterKeeper::class.java, "head")
        setRender(EntityBankerKeeper::class.java, "banker")
        setRender(EntityBarracksKeeper::class.java, "guardelite")
        setRender(EntityDefendArcher::class.java, "hunter")
        setRender(EntityDefendBandit::class.java, "bandit")
        setRender(EntityDefendKnight::class.java, "knight")
        setRender(EntityDefendMage::class.java, "wizard")
        setRender(EntityDefendMarker::class.java, "")
        setRender(EntityDefendPaladin::class.java, "paladin")
        setRender(EntityDefendPriest::class.java, "priest")
        setRender(EntityDefendWarrior::class.java, "warrior")
        setRender(EntityFarmerKeeper::class.java, "inn")
        setRender(EntityFoodKeeper::class.java, "food")
        setRender(EntityLumber::class.java, "foremanlumber")
        setRender(EntityWeaponKeeper::class.java, "smith")
        setRender(EntityGuildMember::class.java, "member")
        setRender(EntityHired::class.java, "guild")
        setRender(EntityForgeKeeper::class.java, "forge")
        setRender(EntityHeadCommander::class.java, "headcommander")
        setRender(EntityKingdomWorker::class.java, "worker")
        setRender(EntityLibraryKeeper::class.java, "librarian")
        setRender(EntityLoneTraveller::class.java, "lone")
        setRender(EntityLostVillager::class.java, "ac")
        setRender(EntityMarkerKeeper::class.java, "")
        setRender(EntityMarker2Keeper::class.java, "")
        setRender(EntityPriestKeeper::class.java, "headpriest")
        setRender(EntityReficulGuardian::class.java, "reficulGuardian")
        setRender(EntityReficulMage::class.java, "reficulMage")
        setRender(EntityReficulSoldier::class.java, "reficulSoldier")
        setRender(EntityShopKeeper::class.java, "forge")
        setRender(EntityStockKeeper::class.java, "stock")
        setRender(EntityTavernKeeper::class.java, "tavern")
        setRender(EntityVillageMember::class.java, "man1")
        setRender(EntityWorkerMember::class.java, "worker")
        setRender(EntityMageKeeper::class.java, "headmage")
        setRender(EntityQuarry::class.java, "foremanquarry")
        setRender(EntityFisher::class.java, "fisher")

        if (isNewYear)
            setRender(EntitySantaClaus::class.java, "santaClaus")
    }

    fun isNewYear(): Boolean
    {
        return isNewYear
    }

    private fun setRegister(entity: Class<out Entity>, name: String)
    {
        EntityRegistry.registerModEntity(entity, name, id++, TaleOfKingdoms.instance, 64, 1, false)
    }

    @SideOnly(Side.CLIENT)
    private fun setRender(entity: Class<out Entity>, location: String)
    {
        RenderingRegistry.registerEntityRenderingHandler(entity, RenderBipedToK(ModelBiped(), 0.4F, location))
    }
}